torch/_dynamo/guards.py

```diff
@@ -569,11 +570,11 @@ def get_sources(t_id, dim):
         for shape_guard in guards:
             self._produce_guard_code(guard, [shape_guard], shape_env=True)
 
-    def TENSOR_MATCH(self, guard: Guard):
+    def TENSOR_MATCH(self, guard: Guard, value=None):
         if guard.is_nn_module():
             self.ID_MATCH(guard)
```

Comment:
numpy ndarray on nn module 🤔  hilarious failure mode

Reply:
This will not fail, but will overspecialze.